packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -161,6 +162,14 @@ export interface BucketDeploymentProps {
    * @see https://docs.aws.amazon.com/AmazonS3/latest/dev/ServerSideEncryptionCustomerKeys.html#sse-c-how-to-programmatically-intro
    */
   readonly serverSideEncryptionCustomerAlgorithm?: string;
+
+  /**
+   * The VPC network whose resources will be available to the AWS Lambda handler
+   * function execution
+   *
+   * @default None
+   */
+  readonly vpc?: ec2.IVpc;
```

Comment:
We normally also add:

```ts
readonly vpcSubnets?: ec2.SubnetSelection;
```

So that users can control exactly which subnets the lambda will be placed in.